gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -214,7 +239,8 @@ private static void setupConjureObjectsProject(
             Supplier<GeneratorOptions> optionsSupplier,
             Task compileConjure,
             TaskProvider<?> compileIrTask,
-            ExtractExecutableTask extractJavaTask) {
+            ArtifactView view,
```

Comment:
you might want to pass a `Provider<File>` instead of constructing it within the method

Reply:
Do you mean in place of the view, or combination of the view + executable name? Unfortunately later in the pipeline we have some special code to apply a ".bat" suffix on windows which could be refactored elsewhere, but makes the change larger.

Reply:
I think we probably want to centralize the logic for windows compat

Reply:
I agree that the windows handling is done poorly, I'd really prefer if that was fixed separately as I don't like to change two things in the same PR.

I do not know what you mean by a `Provider<File>` or how to build one. Gradle provides several types of provider implementations based on files and directories, but I don't see any way to create one from an ArtifactView.

Extracting the executable doesn't need to be aware of the executable script name, the transformation requires the entire extracted tarball which we could later drill down to the executable. We wouldn't want to pull a cached transformation result that neglects to bring jars along with it. Perhaps I've misinterpreted your comment?